colcon_cmake/task/cmake/build.py

```diff
@@ -291,28 +300,41 @@ def _get_make_arguments(self, env):
         :returns: list of make arguments
         :rtype: list of strings
         """
-        # check MAKEFLAGS for -j/--jobs/-l/--load-average arguments
-        makeflags = env.get('MAKEFLAGS', '')
-        regex = (
-            r'(?:^|\s)'
-            r'(-?(?:j|l)(?:\s*[0-9]+|\s|$))'
-            r'|'
-            r'(?:^|\s)'
-            r'((?:--)?(?:jobs|load-average)(?:(?:=|\s+)[0-9]+|(?:\s|$)))'
-        )
-        matches = re.findall(regex, makeflags) or []
-        matches = [m[0] or m[1] for m in matches]
-        if matches:
-            # do not extend make arguments, let MAKEFLAGS set things
-            return []
-        # Use the number of CPU cores
-        jobs = os.cpu_count()
-        with suppress(AttributeError):
-            # consider restricted set of CPUs if applicable
-            jobs = min(jobs, len(os.sched_getaffinity(0)))
-        if jobs is None:
-            # the number of cores can't be determined
-            return []
+        generator = get_generator(args.build_base)
+        if 'Makefiles' in generator and args.cmake_jobs is None:
+            # check MAKEFLAGS for -j/--jobs/-l/--load-average arguments
+            # Note: Ninja does not support environment variables.
+            makeflags = env.get('MAKEFLAGS', '')
+            regex = (
+                r'(?:^|\s)'
+                r'(-?(?:j|l)(?:\s*[0-9]+|\s|$))'
+                r'|'
+                r'(?:^|\s)'
+                r'((?:--)?(?:jobs|load-average)(?:(?:=|\s+)[0-9]+|(?:\s|$)))'
+            )
+            matches = re.findall(regex, makeflags) or []
+            matches = [m[0] or m[1] for m in matches]
+            if matches:
+                # do not extend make arguments, let MAKEFLAGS set things
+                return []
+        # Use command line specified jobs if positive.
+        jobs_arg = args.cmake_jobs if args.cmake_jobs is not None else 0
+        # If positive, use jobs_arg as is, even if it's more than available.
+        # Excessive jobs specified is a user error.
+        jobs = jobs_arg
+        if jobs_arg <= 0:
+            # Base off the number of CPU cores if jobs arg non-positive.
+            jobs = os.cpu_count()
+            with suppress(AttributeError):
+                # consider restricted set of CPUs if applicable
+                jobs = min(jobs, len(os.sched_getaffinity(0)))
+            if jobs is None:
+                # the number of cores can't be determined
+                return []
+            # Finalize jobs as as CPU count deducting the limit specified.
+            jobs = max(jobs + jobs_arg, 1)
+        else:
+            jobs = jobs_arg
```

Comment:
@KazNX nit^2: minor simplification 
```suggestion
        jobs = 0
        if args.cmake_jobs is not None:
            jobs = args.cmake_jobs
        # If positive, use jobs as is, even if it's more than available.
        # Excessive jobs specified is a user error.
        if jobs <= 0:
            # Base off the number of CPU cores if jobs arg non-positive.
            cores = os.cpu_count()
            with suppress(AttributeError):
                # consider restricted set of CPUs if applicable
                cores = min(cores, len(os.sched_getaffinity(0)))
            if cores is None:
                # the number of cores can't be determined
                return []
            # Finalize jobs as as CPU count deducting the limit specified.
            jobs = max(cores + jobs, 1)
```